bot/exts/recruitment/talentpool/_cog.py

```diff
@@ -1,67 +1,81 @@
 import logging
 import textwrap
-from collections import ChainMap
+from collections import ChainMap, defaultdict
 from typing import Union
 
 from discord import Color, Embed, Member, User
 from discord.ext.commands import Cog, Context, group, has_any_role
 
 from bot.api import ResponseCodeError
 from bot.bot import Bot
-from bot.constants import Channels, Guild, MODERATION_ROLES, STAFF_ROLES, Webhooks
+from bot.constants import Guild, MODERATION_ROLES, STAFF_ROLES
 from bot.converters import FetchedMember
-from bot.exts.moderation.watchchannels._watchchannel import WatchChannel
 from bot.exts.recruitment.talentpool._review import Reviewer
 from bot.pagination import LinePaginator
 from bot.utils import time
+from bot.utils.time import get_time_delta
 
 REASON_MAX_CHARS = 1000
 
 log = logging.getLogger(__name__)
 
 
-class TalentPool(WatchChannel, Cog, name="Talentpool"):
+class TalentPool(Cog, name="Talentpool"):
     """Relays messages of helper candidates to a watch channel to observe them."""
 
     def __init__(self, bot: Bot) -> None:
-        super().__init__(
-            bot,
-            destination=Channels.talent_pool,
-            webhook_id=Webhooks.talent_pool,
-            api_endpoint='bot/nominations',
-            api_default_params={'active': 'true', 'ordering': '-inserted_at'},
-            logger=log,
-            disable_header=True,
-        )
-
+        self.bot = bot
         self.reviewer = Reviewer(self.__class__.__name__, bot, self)
         self.bot.loop.create_task(self.reviewer.reschedule_reviews())
 
+        # Stores talentpool users in cache
+        self.cache = defaultdict(dict)
+        self.api_default_params = {'active': 'true', 'ordering': '-inserted_at'}
+
+    async def refresh_cache(self) -> bool:
```

Comment:
This entire function is duplicated in watch channels. Can you also refactor this into a shared function?